cluster/images/etcd/migrate/options.go

```diff
@@ -28,7 +28,7 @@ import (
 )
 
 var (
-	supportedEtcdVersions = []string{"3.0.17", "3.1.12", "3.2.24", "3.3.17", "3.4.13", "3.5.0"}
+	supportedEtcdVersions = []string{"3.0.17", "3.1.12", "3.2.24", "3.3.17", "3.4.13", "3.5.1"}
```

Comment:
append instead of replace?

Reply:
+1 for replacing.

In https://github.com/etcd-io/etcd/blob/main/CHANGELOG-3.5.md, the recommended is 3.5.1+.
> The minimum recommended etcd versions to run in production are 3.2.28+, 3.3.18+, 3.4.2+, and 3.5.1+.

In https://github.com/kubernetes/kubernetes/pull/105706/files#diff-afb644eceb2c0d53027d3e3815f43f489b31b34d18f5402a626f2fbfd4b6eb52R291, kubeadm mark `3.2.18` as the min external etcd version.

Should we remove `"3.0.17", "3.1.12"` here as well? Or the bundled/supported versions should maintain the old versions.

Reply:
+1 for replacing.

In https://github.com/etcd-io/etcd/blob/main/CHANGELOG-3.5.md, the recommended is 3.5.1+.
> The minimum recommended etcd versions to run in production are 3.2.28+, 3.3.18+, 3.4.2+, and 3.5.1+.

In https://github.com/kubernetes/kubernetes/pull/105706/files#diff-afb644eceb2c0d53027d3e3815f43f489b31b34d18f5402a626f2fbfd4b6eb52R291, kubeadm mark `3.2.18` as the min external etcd version.

Should we remove `"3.0.17", "3.1.12"` here as well? Or the bundled/supported versions should maintain the old versions.

Reply:
The list of old etcd vesions in kubeadm is stale. So cleaning them here or separate PR is fine.

Reply:
actually, this list isn't related to kubeadm, right? cc @mborsz for feedback on whether append or replace is correct here

Reply:
doing blame on this line:
https://github.com/kubernetes/kubernetes/blame/e7c3138d94ed4f85073d6d3de543a668c9d60d11/cluster/images/etcd/migrate/options.go#L31

reveals that we always replaced a certain etcd MAJOR.MINOR version with the latest revision or PATCH that we have for the same MINOR.

Reply:
This list is used to decide on which etcd binary to run. Image has multiple binaries to allow handling data on disk from any 3.X version to latest version in image. As Etcd only supports compatibility +/- 1 minor versions we have one version per minor. This means that we should replace "3.5.0" with "3.5.1" as we just need the latest "3.5" binary.